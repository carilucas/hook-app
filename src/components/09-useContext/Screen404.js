import React from 'react'

export const Screen404 = () => {
   return (
      <div>
         <h1 className="text-center mt-5">Error 404</h1>
      </div>
   )
}
