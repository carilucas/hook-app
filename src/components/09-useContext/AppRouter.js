import React from 'react';
import {
BrowserRouter as Router,
   Switch,
   Route,
   Redirect,
} from "react-router-dom";
import { NavBar } from "./NavBar";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { Screen404 } from './Screen404';

export const AppRouter = () => {
   return (
      <Router>
         <div>
            <NavBar />
            <main className="container">
               <Switch>
                  <Route exact path="/" component={ HomeScreen }/>
                  <Route exact path="/about" component= { AboutScreen }/>
                  <Route exact path="/login" component={ LoginScreen }/>
                  <Route path="/not-found" component={ Screen404 }/>
                  <Redirect to="./not-found" />
               </Switch>
            </main>
         </div>
      </Router>
   )
}
