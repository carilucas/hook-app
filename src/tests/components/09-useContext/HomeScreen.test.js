import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas de >HomeScreen/>', () => {
   const user = {
      name:'fernando',
      email:'f@f.com'
   }
   const wrapper = mount(
      <UserContext.Provider value={{
         user
      }}>
       <HomeScreen />
      </UserContext.Provider>
   );
   
   test('Mostrar el componente correctamente snapshot', () => {
      
      expect(wrapper).toMatchSnapshot();

   })
   

})
