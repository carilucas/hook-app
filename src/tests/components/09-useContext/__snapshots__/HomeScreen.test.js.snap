// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de >HomeScreen/> Mostrar el componente correctamente snapshot 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "fernando",
   "email": "f@f.com"
}
  </pre>
</div>
`;
